torch/optim/lr_scheduler.py

```diff
@@ -257,12 +246,9 @@ def __init__(self, optimizer, T_max, eta_min=0, last_epoch=-1):
         super(CosineAnnealingLR, self).__init__(optimizer, last_epoch)
 
     def get_lr(self):
-        if self.last_epoch == 0:
-            return self.base_lrs
-        return [(1 + math.cos(math.pi * self.last_epoch / self.T_max)) /
-                (1 + math.cos(math.pi * (self.last_epoch - 1) / self.T_max)) *
```

Comment:
I did not catch an edge issue here. The induction here only works when the denominator is none-zero. When it is 0, we should use the legacy formula. Please let me know if I should send a fix diff.

Reply:
If you have time to, that would be great! Otherwise I will see it through.

Reply:
No problem, let me do that. Do you want to land this first, and I send a big diff with the fix? Or abandon this and I send a patch?

Reply:
I'll go ahead and wait for you (abandoning when we land your fix). If you end up not having time I'll land this revert before our next release :)

Reply:
See https://github.com/pytorch/pytorch/pull/19180